hpcbench/benchmark/hpl.py

```diff
@@ -174,7 +184,75 @@ def data(self):
             1            # of problems sizes (N)
             ...
         """
-        return self.attributes['data']
+        if self.attributes['data']:
+            return self.attributes['data']
+        else:
+            return self._build_data()
+
+    @property
+    def nodes(self):
+        """used to build HPL.dat"""
+        return self.attributes['nodes']
+
+    @property
+    def cores_per_node(self):
+        """used to build HPL.dat"""
+        return self.attributes['cores_per_node']
+
+    @property
+    def memory_per_node(self):
+        """used to build HPL.dat"""
+        return self.attributes['memory_per_node']
+
+    @property
+    def block_size(self):
+        """used to build HPL.dat"""
+        return self.attributes['block_size']
+
+    def _build_data(self):
+        """Build HPL data from basic parameters"""
+
+        def baseN(nodes, mpn):
+            return int(math.sqrt(mpn * 0.80 * nodes * 1024 * 1024 / 8))
```

Comment:
that's going to be a stupid question.. where does the 0.80 come from?

Reply:
Hello @ohm314 
Thank you for asking.
I ported javascript code used in the webform:
http://www.advancedclustering.com/wp-content/themes/act/js/hpl.js :)